test/integration/test_datatype_upload_dos.py

```diff
@@ -0,0 +1,161 @@
+
+import os
+import string
+import subprocess
+import time
+
+import pytest
+
+from galaxy_test.driver import integration_util
+from .test_datatype_upload import (
+    TEST_CASES,
+    upload_datatype_helper,
+    UploadTestDatatypeDataTestCase
+)
+
+SCRIPT_DIRECTORY = os.path.abspath(os.path.dirname(__file__))
+IRODS_OBJECT_STORE_HOST = os.environ.get('GALAXY_INTEGRATION_IRODS_OBJECT_STORE_HOST', 'localhost')
+IRODS_OBJECT_STORE_PORT = int(os.environ.get('GALAXY_INTEGRATION_IRODS_OBJECT_STORE_PORT', 1247))
+IRODS_OBJECT_STORE_TIMEOUT = int(os.environ.get('GALAXY_INTEGRATION_IRODS_OBJECT_STORE_TIMEOUT', 30))
+IRODS_OBJECT_STORE_USERNAME = os.environ.get('GALAXY_INTEGRATION_IRODS_OBJECT_STORE_USERNAME', 'rods')
+IRODS_OBJECT_STORE_PASSWORD = os.environ.get('GALAXY_INTEGRATION_IRODS_OBJECT_STORE_PASSWORD', 'rods')
+IRODS_OBJECT_STORE_RESOURCE = os.environ.get('GALAXY_INTEGRATION_IRODS_OBJECT_STORE_RESOURCE', 'demoResc')
+IRODS_OBJECT_STORE_ZONE = os.environ.get('GALAXY_INTEGRATION_IRODS_OBJECT_STORE_ZONE', 'tempZone')
+# Run test for only the first 10 test files
+DOS_TEST_CASES = dict(list(TEST_CASES.items())[0:10])
+DISK_OBJECT_STORE_CONFIG = string.Template("""
+<object_store type="distributed">
+    <backends>
+        <backend id="files1" type="disk" weight="1">
+            <files_dir path="${temp_directory}/database/files1"/>
+            <extra_dir type="temp" path="${temp_directory}/database/tmp1"/>
+            <extra_dir type="job_work" path="${temp_directory}/database/job_working_directory1"/>
+        </backend>
+        <backend id="files2" type="disk" weight="1">
+            <files_dir path="${temp_directory}/database/files2"/>
+            <extra_dir type="temp" path="${temp_directory}/database/tmp2"/>
+            <extra_dir type="job_work" path="${temp_directory}/database/job_working_directory2"/>
+        </backend>
+    </backends>
+</object_store>
+""")
+IRODS__OBJECT_STORE_CONFIG = string.Template("""
+<object_store type="distributed">
+    <backends>
+        <backend id="files1" type="disk" weight="1">
+            <files_dir path="${temp_directory}/database/files1"/>
+            <extra_dir type="temp" path="${temp_directory}/database/tmp1"/>
+            <extra_dir type="job_work" path="${temp_directory}/database/job_working_directory1"/>
+        </backend>
+        <backend id="files2" type="irods" weight="1">
+            <auth username="${username}" password="${password}"/>
+            <resource name="${resource}"/>
+            <zone name="${zone}"/>
+            <connection host="${host}" port="${port}" timeout="${timeout}"/>
+            <cache path="${temp_directory}/object_store_cache" size="1000"/>
+            <extra_dir type="job_work" path="${temp_directory}/job_working_directory_irods"/>
+            <extra_dir type="temp" path="${temp_directory}/tmp_irods"/>
+        </backend>
+    </backends>
+</object_store>
+""")
+
+
+def start_irods(container_name):
```

Comment:
This and `stop_irods` looks like it's a duplicate, can you double-check if anything else here can be removed ?

Reply:
Sure, will import from _irods test and delete these.

Reply:
Actually, if I have UploadTestDosIrodsAndDiskTestCase extend IrodsUploadTestDatatypeDataTestCase (per your next comment), I don't need to import the methods at all.